pygmt/helpers/utils.py

```diff
@@ -65,6 +65,8 @@ def data_kind(data, x=None, y=None, z=None):
         raise GMTInvalidInput("Too much data. Use either data or x and y.")
     if data is None and (x is None or y is None):
         raise GMTInvalidInput("Must provided both x and y.")
+    if data is None and required_z and (x is None or y is None or z is None):
```

Comment:
```suggestion
    if data is None and required_z and z is None:
```
I do not think you need to check `x` and `y` again here since this is done before.